circuit/src/layered/circuit.rs

```diff
@@ -293,4 +310,31 @@ impl<C: GKRFieldConfig> Circuit<C> {
             layer.identify_structure_info();
         }
     }
+
+    /// Add a layer before the input layer that contains only relays
+    /// The purpose is to make the input layer contain only addition gates,
+    /// and thus reduces the number of input claims from 2 to 1,
+    /// saving the PCS opening time.
+    pub fn add_input_relay_layer(&mut self) {
+        let input_var_num = self.layers[0].input_var_num;
+        let mut input_relay_layer = CircuitLayer::<C> {
+            input_var_num,
+            output_var_num: input_var_num,
+            ..Default::default()
+        };
+
+        for i in 0..(1 << input_var_num) {
+            input_relay_layer.add.push(GateAdd {
+                i_ids: [i],
+                o_id: i,
+                coef: C::CircuitField::ONE,
+                coef_type: CoefType::Constant,
+                gate_type: 0,
+            });
+        }
+
+        input_relay_layer.structure_info.skip_sumcheck_phase_two = true;
+
+        self.layers.insert(0, input_relay_layer);
+    }
```

Comment:
I see, I somehow realized that, supposing that mul gate exists against input layer, then there does exists 2 PCS opening.

We now introduce a layer with wiring predicate being simply mapping the input to the layer above, then in GKR internal layers (not input layer), the 2 opening claims is randomly merged together with a random linear combination, and thus the "relayed" input layer contains only addition gate with fan in 1, and thus only 1 PCS opening.  I guess I am good with this, am I approaching it in the right direction?